src/routes/jira/connect/enterprise/app/jira-connect-enterprise-app-put.ts

```diff
@@ -1,17 +1,27 @@
-import { NextFunction, Request, Response } from "express";
+import { Request, Response, NextFunction } from "express";
+import { GitHubServerApp } from "~/src/models/github-server-app";
 
 export const JiraConnectEnterpriseAppPut = async (
 	req: Request,
 	res: Response,
 	next: NextFunction
 ): Promise<void> => {
+	req.log.debug("Received Jira Connect Enterprise App PUT request to update app.");
+
 	try {
-		req.log.debug("Received Jira Connect Enterprise App PUT request");
+		const verifiedApp = await GitHubServerApp.getForUuidAndInstallationId(req.params.uuid, res.locals.installation.id);
+
+		if (!verifiedApp || req.params.uuid !== req.body.uuid) {
+			res.status(200).send({ success: false, message: "No GitHub App found. Cannot update." });
+			return next(new Error("No GitHub App found for provided UUID and installationId."));
+		}
 
-		// TODO: Add logic for updating apps
+		await GitHubServerApp.updateGitHubAppByUUID(req.body);
 
-		req.log.debug("Jira Connect Enterprise App updated successfully.", res.locals);
+		res.status(200).send({ success: true });
+		req.log.debug("Jira Connect Enterprise App updated successfully.");
 	} catch (error) {
-		return next(new Error(`Failed to render Jira Connect Enterprise App PUT request: ${error}`));
+		res.status(200).send({ success: false, message: "Failed to update GitHub App." });
+		return next(new Error(`Failed to update GitHub app: ${error}`));
```

Comment:
**IMPORTANT**
I have concern on this, I am not sure how `next` is handling the `new Error`. So worst case is, the variable `error` contains some confidential info, and `next(new Error)` will be captured and render to html somewhere else by some middleware. 

From safety reason, maybe we can omit the `${error}` in the `Error constructor`? But surely we can put a `log.warn` here.

Reply:
The error is handled in error-router `${error}` isn't rendered in prod. If this fails for customers, they will see our standard error page which is rendered in error-router.

Reply:
Fair enough. I am too over concern. All good for this thread.